src/web/controllers/index_controller.py

```diff
@@ -71,3 +72,37 @@ async def favicon(
         return Redirect(
             request.app.route_reverse('static', file_path='/images/sharly-chess.ico')
         )
+
+    @get(
+        path='/404',
+        name='404',
+        cache=1,
+    )
+    async def handle_404(self, request: HTMXRequest) -> HTMXTemplate:
+        web_context: WebContext = WebContext(request)
+
+        return HTMXTemplate(
+            template_name='exceptions/404.html',
```

Comment:
I think exception is developer concept, not a user concept.
```suggestion
            template_name='errors/404.html',
```